### PR TITLE
Attempt to test `fix MkdirAll usage #2361`

### PR DESCRIPTION
**Pull request by rhatdan**
#### What type of PR is this?

> /kind api-change
> /kind bug
> /kind cleanup
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test 
> /kind feature
> /kind flake
> /kind other

#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

**Comment by TomSweeneyRedHat** [comment]
Changes look OK, but the tests are not hip at all.

File: run_linux.go [hunk 1]
```diff
@@ -165,9 +165,28 @@ func (b *Builder) Run(command []string, options RunOptions) error {
 	spec := g.Config
 	g = nil
 
-	logrus.Debugf("ensuring working directory %q exists", filepath.Join(mountPoint, spec.Process.Cwd))
-	if err = os.MkdirAll(filepath.Join(mountPoint, spec.Process.Cwd), 0755); err != nil && !os.IsExist(err) {
-		return err
+	wd := filepath.Join(mountPoint, spec.Process.Cwd)
+	logrus.Debugf("ensuring working directory %q exists", wd)
+	if err = os.MkdirAll(wd, 0755); err != nil {
```
Comment (nalind) [inline run_linux.go:170]: Maybe use `copier.Mkdir(mountPoint, wd, ...)` instead?  It attempts to chroot() to handle symbolic links safely.

Comment (rhatdan) [reply run_linux.go]: Done, PTAL

Comment (nalind) [reply run_linux.go]: Thanks!

**Comment by nalind** [comment]
LGTM

**Comment by TomSweeneyRedHat** [comment]
LGTM
but looks like a rebase is needed.

**Pull request merged**